File: yumi/sdk/dart/lib/src/env.dart
import 'dart:io';

final Map<String, String> _loadedEnv = {};

/// Loads `KEY=value` pairs from a `.env` file into a local map.
/// Does not override existing [Platform.environment] values (same as Go SDK).
void loadEnvFile(String path) {
  final f = File(path);
  if (!f.existsSync()) return;
  for (final line in f.readAsLinesSync()) {
    final t = line.trim();
    if (t.isEmpty || t.startsWith('#')) continue;
    final idx = t.indexOf('=');
    if (idx < 0) continue;
    var key = t.substring(0, idx).trim();
    var value = t.substring(idx + 1).trim();
    if (value.length >= 2) {
      final q = value[0];
      if ((q == '"' || q == "'") && value.endsWith(q)) {
        value = value.substring(1, value.length - 1);
      }
    }
    final existing = Platform.environment[key];
    if (existing == null || existing.isEmpty) {
      _loadedEnv[key] = value;
    }
  }
}

String yumiEnv(String key) {
  final p = Platform.environment[key];
  if (p != null && p.isNotEmpty) return p;
  return _loadedEnv[key] ?? '';
}

void setYumiEnv(String key, String value) {
  _loadedEnv[key] = value;
}

File: yumi/sdk/rust/src/env.rs
use std::env;
use std::fs::File;
use std::io::{BufRead, BufReader};

/// Load a simple `KEY=value` `.env` file. Does not override existing env vars.
pub fn load_env_file(path: &str) {
    let Ok(file) = File::open(path) else {
        return;
    };
    for line in BufReader::new(file).lines().map_while(Result::ok) {
        let line = line.trim();
        if line.is_empty() || line.starts_with('#') {
            continue;
        }
        let Some((key, mut value)) = line.split_once('=') else {
            continue;
        };
        let key = key.trim();
        value = value.trim();
        if value.len() >= 2 {
            let b = value.as_bytes();
            if (b[0] == b'"' && b[value.len() - 1] == b'"')
                || (b[0] == b'\'' && b[value.len() - 1] == b'\'')
            {
                value = &value[1..value.len() - 1];
            }
        }
        if env::var(key).unwrap_or_default().is_empty() {
            env::set_var(key, value);
        }
    }
}
